react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { RxDot, RxDotFilled } from 'react-icons/rx';
+import { HiOutlineChevronLeft, HiOutlineChevronRight } from 'react-icons/hi'
+
+export default function Carousel({ images }) {
+  const [current, setCurrent] = useState(0);
+
+  useEffect(() => {
+    const intervalId = setInterval(handleRightButton, 3000);
+    return () => clearInterval(intervalId);
+  });
```

Comment:
There's no dependency array even though `handleRightButton` is a dependency, so it is called on every render. That will reset your timer on every render.

To use a function in `useEffect` you need to use `useCallback`.